src/app/app.component.ts

```diff
@@ -1,7 +1,25 @@
+import { UserService } from './user.service';
+import { AuthService } from './auth.service';
+import { Router } from '@angular/router';
 import { Component } from '@angular/core';
 
 @Component({
   selector: 'app-root',
   templateUrl: './app.component.html'
 })
-export class AppComponent {}
+export class AppComponent {
+  constructor(
+    private userService: UserService,
+    auth: AuthService,
+    router: Router
+  ) {
+    auth.user$.subscribe(user => {
```

Comment:
when you subscribe to an observable you need to unsubscribe on the onDestroy of that component. Otherwise you will end up leaking memory. This is a topic thats worth reading up on. Let me know if you don't find anything